docs/lang/logic.md

```diff
@@ -0,0 +1,458 @@
+   `e` of `S` that makes the expression `P` evaluate to `FALSE` against the
+   binding `[x |-> e]`.
+
+ _Importantly, when `S = {}`, the expression `\A x \ in S: P` evaluates to
+ `TRUE`, independently of what is written in `P`. Likewise, when `{x \in S: P}
+ = {}`, the expression `\A x \ in S: P` evaluates to `TRUE`._
+   
+**Determinism:** Deterministic.
+
+**Errors:** Pure TLA+ does not restrict the operator arguments.  TLC flags a
+model checking error, if `S` is infinite.  Apalache produces a static type
+error, if the type of elements of `S` is not compatible in the context of `P`
+when an element of `S` is bound to `x`.
+
+**Advanced syntax:** Instead of a single variable `x`, you can use the tuple
+syntax to unpack variables from a Cartesian product, see [Tuples](./tuples.md).
+For instance, one can write `\A <<x, y>> \in S: P`. In this case, for every
+element `e` of `S`, the variable `x` is bound to `e[1]` and `y` is bound to
+`e[2]`. The predicate `P` is evaluated against this binding.
+
+Moreover, instead of introducing one variable, one can quantify over several
+sets. For instance, you can write: `\A x \in S, y \in T: P`. This form is
+simply syntax sugar for the form with nested quantifiers: `\A x \in S: \A y
+\in T: P`.
+
+**Example in TLA+:**
+
+```tla
+  \A x \in {1, 2, 3, 4}:
+    x > 0
+  \* TRUE 
+  \A x \in {1, 2, 3, 4}:
+    x > 2
+  \* FALSE
+  \* check the section on tuples to understand the following syntax
+  \A <<x, y>> \in { 1, 2 } \X { 3, 4 }:
+    x < y
+  \* TRUE
+```
+
+**Example in Python:** Python conveniently offers us a concise syntax:
+
+```python
+  S = frozenset({1, 2, 3, 4})
+  all(x > 0 for x in S)         # True
+  all(x > 2 for x in S)         # False
+  T2 = frozenset((x, y) for x in [1, 2] for y in [3, 4])
+  all(x < y for (x, y) in T2)   # True
+```
+
+----------------------------------------------------------------------------
+
+<a name="existsBounded"></a>
+### Bounded existential quantifier
+
+**Notation:** `\E x \in S: P`
+
+**LaTeX notation:** ![exists-bounded](./img/exists-bounded.png)
+
+**Arguments:** At least three arguments: a variable name, a set, and an
+expression. As usual in TLA+, if the second argument is not a set, the result is
+undefined.You can also use multiple variables and tuples, see **Advanced
+syntax**. 
+
+**Effect:** This operator evaluates to a Boolean value. We explain
+semantics only for a single variable:
+
+ - `\E x \in S: P` evaluates to `TRUE`, if there is an element `e` of `S`
+    that makes the expression `P` evaluate to `TRUE` against the binding
+    `[x |-> e]`.
+ - Conversely, `\E x \in S: P` evaluates to `FALSE`, if for all elements
+   `e` of `S`, the expression `P` evaluate to `FALSE` against the
+   binding `[x |-> e]`.
+
+ _Importantly, when `S = {}`, the expression `\E x \ in S: P` evaluates to
+ `FALSE`, independently of what is written in `P`. Likewise, when `{x \in S: P}
+ = {}`, the expression `\A x \ in S: P` evaluates to `FALSE`._
```

Comment:
This is true, but I don't know why you're giving an example for `\A` in the section on `\E`.

Reply:
It was also wrong. Fixed!